nipype/interfaces/freesurfer/longitudinal.py

```diff
@@ -251,3 +253,200 @@ def _list_outputs(self):
+    >>> baserecon.inputs.subjects_dir = '.'
+    >>> baserecon.cmdline
+    'recon-all -all -base sub-template -sd . -tp ses-1 -tp ses-2'
+    """
+
+    _cmd = "recon-all"
+    input_spec = BaseReconAllInputSpec
+    output_spec = BaseReconAllOutputSpec
+    _can_resume = True
+    force_run = False
+
+    def _gen_subjects_dir(self):
+        return os.getcwd()
+
+
+    def _gen_filename(self, name):
+        if name == "subjects_dir":
+            return self._gen_subjects_dir()
+        return None
+
+
+    def _list_outputs(self):
+        if isdefined(self.inputs.subjects_dir):
+            subjects_dir = self.inputs.subjects_dir
+        else:
+            subjects_dir = self._gen_subjects_dir()
+
+        if isdefined(self.inputs.hemi):
+            hemi = self.inputs.hemi
+        else:
+            hemi = "both"
+
+        outputs = self._outputs().get()
+
+        outputs.update(
+            FreeSurferSource(subject_id=self.inputs.base_id,
+                subjects_dir=subjects_dir, hemi=hemi)._list_outputs()
+        )
+        outputs["subject_id"] = self.inputs.base_id
+        outputs["subjects_dir"] = subjects_dir
+        return outputs
+
+
+    def _is_resuming(self):
+        subjects_dir = self.inputs.subjects_dir
+        if not isdefined(subjects_dir):
+            subjects_dir = self._gen_subjects_dir()
+        if os.path.isdir(os.path.join(subjects_dir, self.inputs.base_id, "mri")):
+            return True
+        return False
+
+
+    def _format_arg(self, name, trait_spec, value):
+        return super(BaseReconAll, self)._format_arg(name, trait_spec, value)
+
+
+    @property
+    def cmdline(self):
+        cmd = super(BaseReconAll, self).cmdline
+
+        if not self._is_resuming():
+            return cmd
+
+        subjects_dir = self.inputs.subjects_dir
+        if not isdefined(subjects_dir):
+            subjects_dir = self._gen_subjects_dir()
+
+        directive = self.inputs.directive
+        if not isdefined(directive):
+            steps = []
+
+        iflogger.info(f"recon-all: {cmd}")
+        return cmd
+
+
+class LongReconAllInputSpec(ReconAllInputSpec):
+    subject_id = traits.Str(argstr="-subjid %s", desc="subject name")
+    long_id = traits.Tuple(
+        traits.Str(),
+        traits.Str(),
+        argstr="-long %s %s",
+        desc="longitudinal name followed by base template name",
+        xor=["subject_id"]
+    )
+
+
+class LongReconAllOutputSpec(FreeSurferSource.output_spec):
+    subjects_dir = Directory(exists=True, desc="FreeSurfer subjects directory")
+    subject_id = traits.Str(desc="Subject template name")
+
+
+class LongReconAll(CommandLine):
+    """Uses FreeSurfer's longitudinal recon-all to process a subject given
+    the previously processed base template.
+
+    Examples
+    ---------
+
+    >>> from nipype.interfaces.freesurfer.longitudinal import LongReconAll
+    >>> longrecon = LongReconAll()
+    >>> longrecon.inputs.long_id = ("ses-1","sub-template")
```

Comment:
From a workflow construction perspective, providing `long_id` as tuple might make composition from `all`, `base` and then `long` interfaces more difficult.

In [pydra-freesurfer](https://github.com/aramis-lab/pydra-freesurfer/blob/main/pydra/tasks/freesurfer/recon_all.py), I have split it as `longitudinal_timepoint_id` and `longitudinal_template_id` and required the latter be specified if the former is. This way they are both exposed as separate input to the interface.

Reply:
Oh, that's a very good point. I can separate those out. I tend to run each step separately, but I will make that change and test out a full workflow with all three steps when I can...should definitely make sure that works with this setup.